File: 2015/Day5/Day5.py
niceStrings = 0

with open("2015\Day5\input.txt") as input:
    lines = input.readlines()

    for i, line in enumerate(lines):
        if "ab" in line or "cd" in line or "pq" in line or "xy" in line:
            continue
        
        for ch in line:
            if ch + ch in line:
                break
        else:
            continue

        nVowels = 0

        for ch in line:
            if ch is 'a' or ch is 'e' or ch is 'i' or ch is 'o' or ch is 'u':
                nVowels += 1
        
        if not(nVowels > 2):
            continue

        print(line)
        niceStrings += 1

print(niceStrings)

File: 2015/Day5/day5.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main() {
    ifstream cin("input.txt");

    string line;
    int nice_strings = 0;

    while(!cin.eof()) {
        cin >> line;

        if (line.find("ab") < line.length() || line.find("cd") < line.length() || line.find("pq") < line.length() || line.find("xy") < line.length()) 
            continue;

        bool brk = true;
        for (char& ch : line) {
            string tmp = string() + ch + ch;
            if (line.find(tmp) < line.length()) {
                brk = false;
                break;
            }
        }

        if (brk) continue;

        int n_vowels = 0;

        for (char& ch : line) {
            if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u')
                n_vowels++;
        }

        if (!(n_vowels > 2))
            continue;

        nice_strings++;
    }

    cout << nice_strings;
}